### PR TITLE
Add cluster architecture info

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary
Fixes https://github.com/rancher/dashboard/issues/10382

### Occurred changes and/or fixed issues

Adding cluster `architecture` info in various places:

- Homepage, Clusters table
  - adding 'Distro' sub-label to Provider column.
  - adding 'Architecture' sub-label to Kubernetes Version column.
  - adding architecture info in Kubernetes Version cells
- Cluster Management, Clusters table
  - same
- Cluster Dashboard, for each cluster
  - adding the new 'Architecture' on top of the page.

### Technical notes summary

- `SortableTable` component: fixed css -> header's alignment for tables containing columns with `subLabel`.

### Areas or cases that should be tested

- Clusters tables

### Areas which could experience regressions

- SortableTable component

### Screenshot/Video

Homepage

image

Cluster Management

image

Cluster Dashboard (local)

image

Tooltips

- mixed

image

image

- mixed, one is unknown

image

image

- all are unknown

image

image

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by momesgin** [comment]
I created a DO cluster choosing an intel system from the Size dropdown, but 'Amd64' is displayed under the architecture column:

Screenshot 2024-04-02 at 4 05 19 PM


image

**Comment by torchiaf** [comment]
> I created a DO cluster choosing an intel system from the Size dropdown, but 'Amd64' is displayed under the architecture column:
> 

Although it may seem weird, intel processors are labeled as `amd64` . UI side, it has to be consistent with the value in `kubernetes.io/arch` label.

**Comment by momesgin** [comment]
> > I created a DO cluster choosing an intel system from the Size dropdown, but 'Amd64' is displayed under the architecture column:
> 
> Although it may seem weird, intel processors are labeled as `amd64` . UI side, it has to be consistent with the value in `kubernetes.io/arch` label.

That's so weird! thanks for explaining

**Comment by torchiaf** [comment]
@momesgin I pushed a re-base to fix the conflicts with `master`.

File: shell/models/provisioning.cattle.io.cluster.js [hunk 1]
```diff
@@ -403,6 +404,29 @@ export default class ProvCluster extends SteveModel {
...
+    });
+
+    return obj;
+  }
+
+  get architecture() {
+    const keys = Object.keys(this.nodesArchitecture);
+
+    return {
+      label:   keys.length === 1 ? keys[0] : 'Mixed',
```
[the remark this answers is not in the document]
Comment (torchiaf) [reply shell/models/provisioning.cattle.io.cluster.js]: Thanks @momesgin , I applied the changes. Please, have a look to the screenshots in the description.

**Review by momesgin — APPROVED** [review]
LGTM

**Pull request merged**